src/LayerEditor/ui/data/le_model.py

```diff
@@ -98,41 +127,94 @@ def confront_file_timestamp(self):
                 pass
         return False
 
-    def add_region(self, name, dim):
-        """ Add new region according to the provided data from the current tab and select it."""
+    def add_region(self, reg):
         with undo.group("Add new region"):
-            reg = RegionItem(name=name, dim=dim)
             self.regions_model.add_region(reg)
             self.gui_curr_block.gui_selected_layer.set_gui_selected_region(reg)
-        self.region_list_changed.emit()
-        self.invalidate_scene.emit()
+            self.emit_region_list_changed()
+            self.emit_invalidate_scene()
         return reg
 
     def delete_region(self, reg):
-        with undo.group("Add new region"):
+        with undo.group("Delete new region"):
             self.regions_model.delete_region(reg)
-            for block in self.blocks_model.blocks.values():
-                for layer in block.layers:
-                    if layer.gui_selected_region == reg:
-                        layer.set_gui_selected_region(RegionItem.none)
-        self.region_list_changed.emit()
-        self.invalidate_scene.emit()
-
-    def get_curr_layer_index(self):
-        return self.gui_curr_block.layers.index(self.gui_curr_block.gui_selected_layer)
+            self.emit_region_list_changed()
+            self.emit_invalidate_scene()
 
     def is_region_used(self, reg):
-        dim = self.regions_model.regions[reg].dim
+        """Returns True if reg is assigned to any shape in any layer in any block"""
         for block in self.blocks_model.blocks.values():
-            for layer in block.layers:
+            for layer in block.layers_dict.values():
+                dim = self.regions_model.regions[reg].dim
                 if layer.is_stratum:
                     dim -= 1
                     if dim < 0:
                         continue
+                else:
+                    if dim == 3:
+                        continue
                 if reg in layer.shape_regions[dim].values():
                     return True
         return False
 
+    def delete_layer_top(self, layer):
+        """ Delete specified layer and top interface.
+            Layer above deleted layer (if it exists) will be extended to bottom interface of deleted layer.
+            There must be at least one stratum layer above specified layer in parameter(fracture layer doesn't count)"""
+        with undo.group("Delete layer and extend the layer above"):
+            layers = layer.block.get_sorted_layers()
+            idx = layers.index(layer)
+            if idx != 0 or (idx == 1 and layers[idx - 1].is_stratum):
+                if layers[idx - 1].is_stratum:
+                    above_layer = layers[idx - 1]
+                else:
+                    layer.block.delete_layer(layers[idx - 1])
```

Comment:
That was intended behviour, since user is deleting layer and interface, since the fracture layer is on the interface it makes sense to me to delete the fracture layer as well. But I can add a warning message. Not sure how to delete the interface while preserving the fracture layer.